sdk/src/lib.rs

```diff
@@ -95,6 +95,7 @@ pub mod reward_info;
 pub mod reward_type;
 pub mod rpc_port;
 pub mod secp256k1_instruction;
+pub mod secp256r1_instruction;
```

Comment:
Please just make this a new crate, as solana-sdk is already extremely bloated

Reply:
Maybe I'm misunderstanding but `sdk/src/precompiles.rs` determines the structure and definition of the existing precompiles. If i were to move `secp256r1_program.rs` and `secp256r1_instruction.rs` into their own crate, sdk would still have that crate as a dep, no?

Reply:
You would be correct but work is already in progress to move precompiles.rs out of the SDK. So everything will be easier if your work is in its own crate from day one

Reply:
Got it 👍

Reply:
Would you mind adding the `CI` flag? Ive run the CI locally a couple times but it takes ages and slows my system down :|

Reply:
> Please just make this a new crate, as solana-sdk is already extremely bloated

Locally refactored everything into its own `secp256r1` crate. Only things missing are the extraction of `PrecompileError` [(looks like its almost complete)](https://github.com/anza-xyz/agave/pull/2300) and the subsequent extraction of `solana_sdk::transaction`. 

Alternate solution if `solana_sdk::transaction` is too much of a hassle for now, is to move the [one integration test](https://github.com/anza-xyz/agave/blob/6011adf9350a4d18661b29b8820d2133b211f2a2/sdk/src/secp256r1_instruction.rs#L481) depending on `transaction` into the `precompiles.rs` file. That way `secp256r1` can avoid the `solana_sdk::transaction` dependency all-together.

Reply:
oh if we can get to a place where solana-sdk is only in dev-dependencies that's fine. Circularity in dev-dependencies is no problem (just need to declare them with `{ path = ` instead of `{ workspace = true }` or `cargo publish` gets upset)

Reply:
Oh sweet! Will pull in the error extraction (just saw it merged 🙌), refactor `solana_sdk::transaction` as a dev-dep and then push 🫡

Reply:
Extracted it in [86a25ea](https://github.com/anza-xyz/agave/pull/3152/commits/86a25ea997b9c52b54a03997f1d7e2c74a7e2464) 👍  Let me know if you have any nits or feedback.

Would also appreciate another CI run 🫡 

Sidenote: Cargo seems to constantly modify `version = 3` to `version = 4` in the `sbf/Cargo.lock` file. This in turn makes one of the sub-tasks in `ci/run-local.sh` error out. Any chance you know of a fix?